sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -616,7 +619,8 @@ private[hive] class HiveClientImpl(
     try {
       Option(client.getFunction(db, name)).map(fromHiveFunction)
     } catch {
-      case he: HiveException => None
+      case CausedBy(ex: NoSuchObjectException) if ex.getMessage.contains(name) =>
```

Comment:
In my case, the Exception thrown is "org.apache.hadoop.hive.ql.metadata.HiveException:MetaException(message:NoSuchObjectException(message:Function default.myfunc does not exist))", but it turns out that the root case of this exception is MetaException whose message is "NoSuchObjectException(message:Function default.myfunc does not exist))", thus the exception is not caught. (I run into this problem when I use "CREATE FUNCTION" using spark sql with hive)